### ISSUE TITLE
[TTIRFusing] Support SDPA fusing with ttir.where after ttir.softmax

### ISSUE THREAD

**Issue by jserbedzijaTT** [tenstorrent/tt-mlir#6140]
### Describe the bug

The following test case is not fused to ttir.scaled_dot_product_attention:
```
module @SyncTensorsGraph.83 attributes {mhlo.cross_program_prefetches = [], mhlo.input_output_alias = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false, ttcore.meshes = #ttcore.meshes<[<"mesh" = 1x1>]>} {
  ttcore.device_module {
    builtin.module @SyncTensorsGraph.83 attributes {mhlo.cross_program_prefetches = [], mhlo.input_output_alias = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false, ttcore.meshes = #ttcore.meshes<[<"mesh" = 1x1>]>} {
      func.func @main(%arg0: tensor<1x1x4096x512xbf16> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.shard_status = #ttcore.shard_status<unsharded>, ttir.name = "args_0"} loc("xla__device_data"), %arg1: tensor<1x1x4096x512xbf16> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.shard_status = #ttcore.shard_status<unsharded>, ttir.name = "args_1"} loc("xla__device_data"), %arg2: tensor<1x1x4096x512xbf16> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.shard_status = #ttcore.shard_status<unsharded>, ttir.name = "args_2"} loc("xla__device_data")) -> (tensor<1x1x4096x512xbf16> {ttcore.shard_status = #ttcore.shard_status<unsharded>}) {
        %0 = "ttir.constant"() <{value = dense<true> : tensor<i1>}> : () -> tensor<i1> loc(#loc)
        %1 = "ttir.constant"() <{value = dense<false> : tensor<i1>}> : () -> tensor<i1> loc(#loc)
        %2 = "ttir.constant"() <{value = dense<0xFF800000> : tensor<f32>}> : () -> tensor<f32> loc(#loc)
        %3 = "ttir.constant"() <{value = dense<0.210224107> : tensor<f32>}> : () -> tensor<f32> loc(#loc)
        %4 = "ttir.constant"() <{value = dense<0xFFF0000000000000> : tensor<f64>}> : () -> tensor<f64> loc(#loc)
        %5 = "ttir.constant"() <{value = dense<0.000000e+00> : tensor<f32>}> : () -> tensor<f32> loc(#loc)
        %6 = "ttir.reshape"(%5) <{shape = [1 : i32, 1 : i32, 1 : i32, 1 : i32]}> : (tensor<f32>) -> tensor<1x1x1x1xf32> loc(#loc)
        %7 = "ttir.broadcast"(%6) <{broadcast_dimensions = array<i64: 1, 1, 4096, 4096>}> : (tensor<1x1x1x1xf32>) -> tensor<1x1x4096x4096xf32> loc(#loc)
        %8 = "ttir.reshape"(%4) <{shape = [1 : i32, 1 : i32, 1 : i32, 1 : i32]}> : (tensor<f64>) -> tensor<1x1x1x1xf64> loc(#loc)
        %9 = "ttir.broadcast"(%8) <{broadcast_dimensions = array<i64: 1, 1, 4096, 4096>}> : (tensor<1x1x1x1xf64>) -> tensor<1x1x4096x4096xf64> loc(#loc)
        %10 = "ttir.reshape"(%3) <{shape = [1 : i32, 1 : i32, 1 : i32, 1 : i32]}> : (tensor<f32>) -> tensor<1x1x1x1xf32> loc(#loc)
        %11 = "ttir.broadcast"(%10) <{broadcast_dimensions = array<i64: 1, 1, 512, 4096>}> : (tensor<1x1x1x1xf32>) -> tensor<1x1x512x4096xf32> loc(#loc)
        %12 = "ttir.reshape"(%3) <{shape = [1 : i32, 1 : i32, 1 : i32, 1 : i32]}> : (tensor<f32>) -> tensor<1x1x1x1xf32> loc(#loc)
        %13 = "ttir.broadcast"(%12) <{broadcast_dimensions = array<i64: 1, 1, 4096, 512>}> : (tensor<1x1x1x1xf32>) -> tensor<1x1x4096x512xf32> loc(#loc)
        %14 = "ttir.typecast"(%arg2) <{conservative_folding = false}> : (tensor<1x1x4096x512xbf16>) -> tensor<1x1x4096x512xf32> loc(#loc2)
        %15 = "ttir.multiply"(%14, %13) : (tensor<1x1x4096x512xf32>, tensor<1x1x4096x512xf32>) -> tensor<1x1x4096x512xf32> loc(#loc3)
        %16 = "ttir.reshape"(%15) <{shape = [1 : i32, 4096 : i32, 512 : i32]}> : (tensor<1x1x4096x512xf32>) -> tensor<1x4096x512xf32> loc(#loc4)
        %17 = "ttir.typecast"(%arg1) <{conservative_folding = false}> : (tensor<1x1x4096x512xbf16>) -> tensor<1x1x4096x512xf32> loc(#loc2)
        %18 = "ttir.permute"(%17) <{permutation = array<i64: 0, 1, 3, 2>}> : (tensor<1x1x4096x512xf32>) -> tensor<1x1x512x4096xf32> loc(#loc5)
        %19 = "ttir.multiply"(%18, %11) : (tensor<1x1x512x4096xf32>, tensor<1x1x512x4096xf32>) -> tensor<1x1x512x4096xf32> loc(#loc3)
        %20 = "ttir.reshape"(%19) <{shape = [1 : i32, 512 : i32, 4096 : i32]}> : (tensor<1x1x512x4096xf32>) -> tensor<1x512x4096xf32> loc(#loc4)
        %21 = "ttir.dot_general"(%16, %20) <{batch_dims_lhs = array<i64: 0>, batch_dims_rhs = array<i64: 0>, contract_dims_lhs = array<i64: 2>, contract_dims_rhs = array<i64: 1>}> : (tensor<1x4096x512xf32>, tensor<1x512x4096xf32>) -> tensor<1x4096x4096xf32> loc(#loc6)
        %22 = "ttir.reshape"(%21) <{shape = [1 : i32, 1 : i32, 4096 : i32, 4096 : i32]}> : (tensor<1x4096x4096xf32>) -> tensor<1x1x4096x4096xf32> loc(#loc4)
        %23 = "ttir.typecast"(%22) <{conservative_folding = false}> : (tensor<1x1x4096x4096xf32>) -> tensor<1x1x4096x4096xf64> loc(#loc7)
        %24 = "ttir.eq"(%23, %9) : (tensor<1x1x4096x4096xf64>, tensor<1x1x4096x4096xf64>) -> tensor<1x1x4096x4096xi1> loc(#loc7)
        %25 = "ttir.logical_not"(%24) : (tensor<1x1x4096x4096xi1>) -> tensor<1x1x4096x4096xi1> loc(#loc8)
        %26 = "ttir.reduce_or"(%25) <{dim_arg = [3 : i32], keep_dim = false}> : (tensor<1x1x4096x4096xi1>) -> tensor<1x1x4096xi1> loc(#loc9)
        %27 = "ttir.reshape"(%26) <{shape = [1 : i32, 1 : i32, 4096 : i32, 1 : i32]}> : (tensor<1x1x4096xi1>) -> tensor<1x1x4096x1xi1> loc(#loc9)
        %28 = "ttir.logical_not"(%27) : (tensor<1x1x4096x1xi1>) -> tensor<1x1x4096x1xi1> loc(#loc8)
        %29 = "ttir.reshape"(%28) <{shape = [1 : i32, 1 : i32, 4096 : i32]}> : (tensor<1x1x4096x1xi1>) -> tensor<1x1x4096xi1> loc(#loc10)
        %30 = "ttir.reshape"(%29) <{shape = [1 : i32, 1 : i32, 4096 : i32, 1 : i32]}> : (tensor<1x1x4096xi1>) -> tensor<1x1x4096x1xi1> loc(#loc10)
        %31 = "ttir.broadcast"(%30) <{broadcast_dimensions = array<i64: 1, 1, 1, 4096>}> : (tensor<1x1x4096x1xi1>) -> tensor<1x1x4096x4096xi1> loc(#loc10)
        %32 = "ttir.max"(%22) <{dim_arg = [3 : i32], keep_dim = false}> : (tensor<1x1x4096x4096xf32>) -> tensor<1x1x4096xf32> loc(#loc11)
        %33 = "ttir.reshape"(%32) <{shape = [1 : i32, 1 : i32, 4096 : i32, 1 : i32]}> : (tensor<1x1x4096xf32>) -> tensor<1x1x4096x1xf32> loc(#loc11)
        %34 = "ttir.broadcast"(%33) <{broadcast_dimensions = array<i64: 1, 1, 1, 4096>}> : (tensor<1x1x4096x1xf32>) -> tensor<1x1x4096x4096xf32> loc(#loc11)
        %35 = "ttir.subtract"(%22, %34) : (tensor<1x1x4096x4096xf32>, tensor<1x1x4096x4096xf32>) -> tensor<1x1x4096x4096xf32> loc(#loc11)
        %36 = "ttir.exp"(%35) : (tensor<1x1x4096x4096xf32>) -> tensor<1x1x4096x4096xf32> loc(#loc11)
        %37 = "ttir.sum"(%36) <{dim_arg = [3 : i32], keep_dim = false}> : (tensor<1x1x4096x4096xf32>) -> tensor<1x1x4096xf32> loc(#loc11)
        %38 = "ttir.reshape"(%37) <{shape = [1 : i32, 1 : i32, 4096 : i32, 1 : i32]}> : (tensor<1x1x4096xf32>) -> tensor<1x1x4096x1xf32> loc(#loc11)
        %39 = "ttir.broadcast"(%38) <{broadcast_dimensions = array<i64: 1, 1, 1, 4096>}> : (tensor<1x1x4096x1xf32>) -> tensor<1x1x4096x4096xf32> loc(#loc11)
        %40 = "ttir.div"(%36, %39) : (tensor<1x1x4096x4096xf32>, tensor<1x1x4096x4096xf32>) -> tensor<1x1x4096x4096xf32> loc(#loc11)
        %41 = "ttir.where"(%31, %7, %40) : (tensor<1x1x4096x4096xi1>, tensor<1x1x4096x4096xf32>, tensor<1x1x4096x4096xf32>) -> tensor<1x1x4096x4096xf32> loc(#loc12)
        %42 = "ttir.reshape"(%41) <{shape = [1 : i32, 4096 : i32, 4096 : i32]}> : (tensor<1x1x4096x4096xf32>) -> tensor<1x4096x4096xf32> loc(#loc4)
        %43 = "ttir.typecast"(%arg0) <{conservative_folding = false}> : (tensor<1x1x4096x512xbf16>) -> tensor<1x1x4096x512xf32> loc(#loc2)
        %44 = "ttir.reshape"(%43) <{shape = [1 : i32, 4096 : i32, 512 : i32]}> : (tensor<1x1x4096x512xf32>) -> tensor<1x4096x512xf32> loc(#loc4)
        %45 = "ttir.dot_general"(%42, %44) <{batch_dims_lhs = array<i64: 0>, batch_dims_rhs = array<i64: 0>, contract_dims_lhs = array<i64: 2>, contract_dims_rhs = array<i64: 1>}> : (tensor<1x4096x4096xf32>, tensor<1x4096x512xf32>) -> tensor<1x4096x512xf32> loc(#loc6)
        %46 = "ttir.reshape"(%45) <{shape = [1 : i32, 1 : i32, 4096 : i32, 512 : i32]}> : (tensor<1x4096x512xf32>) -> tensor<1x1x4096x512xf32> loc(#loc4)
        %47 = "ttir.typecast"(%46) <{conservative_folding = false}> : (tensor<1x1x4096x512xf32>) -> tensor<1x1x4096x512xbf16> loc(#loc2)
        return %47 : tensor<1x1x4096x512xbf16> loc(#loc)
      } loc(#loc)
    } loc(#loc)
  } loc(#loc)
} loc(#loc)
#loc = loc(unknown)
#loc2 = loc("xla__cast")
#loc3 = loc("aten__mul")
#loc4 = loc("aten__view")
#loc5 = loc("aten__permute")
#loc6 = loc("aten__matmul")
#loc7 = loc("aten__eq")
#loc8 = loc("aten__logical_not")
#loc9 = loc("aten__any")
#loc10 = loc("aten__expand")
#loc11 = loc("aten__softmax")
#loc12 = loc("aten__where")
```

### Expected behavior

Fuse the test to sdpa

**Comment by ppadjinTT** [comment]
Context: This is the origin of this TTIR graph:
```
import torch
import torch.nn as nn
import torch_xla.core.xla_model as xm
import torch_xla.runtime as xr

def run_sdpa():
    xr.set_device_type("TT")
    import torch.nn.functional as F
    
    class SDPA(nn.Module):
        def __init__(self):
            super().__init__()
        def forward(self, query, key, value):
            return F.scaled_dot_product_attention(query, key, value, attn_mask=None, dropout_p=0.0, is_causal=False)

    query = torch.randn(1, 1, 4096, 512, dtype=torch.bfloat16)
    key = torch.randn(1, 1, 4096, 512, dtype=torch.bfloat16)
    value = torch.randn(1, 1, 4096, 512, dtype=torch.bfloat16)

    model = SDPA()
    model = model.to(torch.bfloat16)
    model = model.eval()
    model.compile(backend='tt')
    device = xm.xla_device()
    model = model.to(device)
    query = query.to(device)
    key = key.to(device)
    value = value.to(device)

    model.eval()
    with torch.no_grad():
        tt_output = model(query, key, value)
    tt_output_cpu = tt_output.cpu()
    print(tt_output_cpu)

if __name__ == "__main__":
    run_sdpa()
```

This means that a ` F.scaled_dot_product_attention` operation is not successfully fused inside MLIR. Our current hypothesis is that the dropout element in this operation generates a `ttir.where` operation in MLIR graph, which is not currently accounted for
